sway-core/src/semantic_analysis/ast_node/declaration/trait.rs

```diff
@@ -627,7 +630,9 @@ impl TyTraitDecl {
             Some(self.span()),
             IsImplSelf::No,
             IsExtendingExistingImpl::No,
+            IsImplInterfaceSurface::Yes,
         );
+        assert!(!interface_handler.has_errors());
```

Comment:
This assert! will crash the compiler (and LSP) in production if interface_handler.has_errors() returns true. Can we avoid this?

Reply:
It's a simple question for me: if it's possible that this leads to semantically invalid binaries, it should be an ICE, if not it can be a debug assert for our debugging benefit.